contrib/git-list-between

```diff
@@ -9,32 +9,58 @@ print_help() {
     echo 'Usage: git-list-between BRANCH YYYY/MM/DD YYYY/MM/DD [[--] <path>...]'
 }
 
+# Workaround for https://github.com/elfshaker/elfshaker/issues/89, that
+# there are occasional issues with the clock, and '--since'
+# unfortunately stops once it sees the first date which is older than
+# the specified time. Strategy: Go back a large number of commits, and
+# find the earliest with the given date.
+find-first-commit() {
+    ARGS=(
+        # Tradeoff of runtime vs max expected commits in the month.
+        # It's >10x higher than I have observed in a month.
+        --max-count 40000
+        --first-parent
+        --topo-order
+        --until="$DATETIME_UPPER"
+        --no-commit-header
+        --format='%cd %H'
+        --date=format-local:"%Y/%m/%d"
+        "$BRANCH"
+    )
+
+    TZ=UTC git rev-list "${ARGS[@]}" | \
+        egrep "^$DATE_LOWER" | \
```

Comment:
Do we assume that there is a commit with that date? And if there is not, what output do we see? Seemingly `git log ..${BRANCH}` from `list()` which is same as `git log HEAD..${BRANCH}`.

Reply:
Good catch. Guess this could theoretically break on January 1st as an example. Though in practice I think there is always something getting committed to LLVM. Any smart ideas to address that? I might be better off making it month specific rather than date specific 🤔. Not great in any case.

Reply:
We've assumed that CommitDates are correct, but actually a commit with a future or past date can pop into the history at any point, although I concur that a clock being ahead is less likely.

`git log --since 2021/09/01 --until 2021/10/01 --topo-order --format="%cd %H"`
Now includes the commit made in August 2022. 

I think the best approach is to enter commit SHAs (ranges) by hand in the future, using our human understanding of git history. Unless we can come up with a clever algorithm that does that for us and does not skip commits on the boundary.

Reply:
Thanks for continuing the thinking, indeed skew-to-future could also be a problem and I also can't see an easy way around it. Even the human chosen ranges could be wrong since the method they use to find appropriate commits may rely on this same machinery. The main safety check is either a count of commits by day within a month or more crudely checking the count of commits within the month. Another thing is that assuming we're typically building the most recent pack promptly, we have the more concrete data point that the history just fetched can't contain much 'junk into the future' since it doesn't exist yet.

I'll have to come back to this and do a little bit more thinking, since I want to automate it and leave it running for a long while.